src/utils/convertPartialDateStringToObj.ts

```diff
@@ -0,0 +1,72 @@
+import { DateObjectSchema } from 'serina.schema';
+import { DATES, PARTIAL_DATES } from 'filters/dates/dates.constants';
+import contains from './contains';
+import { monthStringToNumber } from 'utils';
+import { DateTime } from 'luxon';
+
+function strToInt(dayStr: string, monthStr: string, yearStr: string): DateObjectSchema {
+    return {
+        day: parseInt(dayStr, 10),
+        month: parseInt(monthStr, 10),
+        year: parseInt(yearStr, 10),
+    };
+}
+
+/**
+ * We want to return a future date, so if the month has already occurred this year, we give next year's date.
+ */
+function getYearWhenMonthIsUpcoming(month): string {
+    const currentDate = DateTime.utc();
+    return (parseInt(month, 10) > currentDate.month ? currentDate.year : currentDate.year + 1).toString();
```

Comment:
What if today's date is `25th Aug`, I typed `24th Aug`. I think you also need to take day into consideration.

Reply:
I don't see a use case for this. It makes sense in a situation where you are in, e.g. December and planning to do something for 1st of January. Clearly you probably mean 1st of January next year. But would you really use a partial date for something that is a year away? Most likely that would refer to yesterday, rather than the date a full year away.

Reply:
I think if the user type a date in the past (not just year), it should assume next year by default.

Example: today is 25th Aug

I type 30th July, I get 2020.
I type 1st Aug, I get 2019

If date is in the past, then next year ELSE this year. I think this is clearer than if you type month then we may change the year you are talking about, but we will ignore day because we assume you won't do it. That just makes no sense. It needs to be more logic.

Reply:
Okay fair enough, I guess there is no perfect solution here.